napari/_qt/_qapp_model/qactions/__init__.py

```diff
@@ -0,0 +1,31 @@
+from functools import lru_cache
+from typing import Optional
+
+# Submodules should be able to import from most modules, so to
+# avoid circular imports, don't import submodules at the top level here,
+# import them inside the init_qactions function.
+
+
+@lru_cache  # only call once
+def init_qactions():
+    from itertools import chain
+
+    from napari._app_model import get_app
+
+    from ...qt_main_window import Window, _QtMainWindow
+    from ._window import WINDOW_ACTIONS
+
+    # Qt-specific providers/processors
+    def _provide_window() -> Optional[Window]:
+        if _qmainwin := _QtMainWindow.current():
+            return _qmainwin._window
```

Comment:
this should be somehow tested

Reply:
I totally agree, again this was copied over from #4826. I think we have been working with the idea that #4826 would be the main PR that provides the base changes required to migrate these menus to app-model, and should be merged first. Note that there are 4 other PRs that follow on from #4826 and use it as a template (see first item in https://github.com/napari/napari/issues/4860#issue-1315990449). To keep things clean it might be a good idea to not spread the base changes (which have been copied to all the other PRs) and tests across the many PRs and keep it just in the first one? I will note this down in https://github.com/napari/napari/issues/4860 and here but happy to do it a different way if you prefer?

You are correct in that this (and a few other functions) are not tested in #4826.

Reply:
@lucyleeow I made a comment in #4826 about how we could maybe test this and the viewer provider.